hadoop/Dockerfile

```diff
@@ -48,6 +36,40 @@ RUN curl --fail "https://repo.stackable.tech/repository/packages/jmx-exporter/jm
     ln -s "/stackable/jmx/jmx_prometheus_javaagent-${JMX_EXPORTER}.jar" /stackable/jmx/jmx_prometheus_javaagent.jar && \
     ln -s /stackable/jmx/jmx_prometheus_javaagent.jar /stackable/jmx/jmx_prometheus_javaagent-0.16.1.jar
 
+# This Protobuf version is the exact version as used in the Hadoop Dockerfile
+# See https://github.com/apache/hadoop/blob/trunk/dev-support/docker/pkg-resolver/install-protobuf.sh
+# (this was hardcoded in the Dockerfile in earlier versions of Hadoop, make sure to look at the exact version in Github)
+# For now all versions of Hadoop we support use Protobuf 3.7.1 so we can hardcode it here.
+# Should it ever differ between versions we'll need to make this a variable as well.
+RUN mkdir -p /opt/protobuf-src && \
+    curl --fail -L -s -S https://repo.stackable.tech/repository/packages/protobuf/protobuf-java-3.7.1.tar.gz -o /opt/protobuf.tar.gz && \
+    tar xzf /opt/protobuf.tar.gz --strip-components 1 -C /opt/protobuf-src --no-same-owner && \
+    cd /opt/protobuf-src && \
+    ./configure --prefix=/opt/protobuf && \
+    make "-j$(nproc)" && \
+    make install && \
+    cd /root && \
+    rm -rf /opt/protobuf-src
+
+ENV PROTOBUF_HOME /opt/protobuf
+ENV PATH "${PATH}:/opt/protobuf/bin"
+
+RUN curl --fail -L https://repo.stackable.tech/repository/packages/node/node-v${NODE}-linux-x64.tar.xz | tar -xJC . && \
+    ln -s /stackable/node-v${NODE}-linux-x64 /stackable/node
+ENV PATH "${PATH}:/stackable/node/bin"
+RUN corepack enable yarn && \
+    yarn config set ignore-engines true
+
+RUN cd /stackable/hadoop-${PRODUCT}-src && \
+    # Hadoop Pipes requires libtirpc to build, whose headers are not packaged in RedHat UBI
+    git apply < /stackable/patches/0001-disable-pipes.patch && \
+    # Datanode registration override is required for Listener Operator integration
+    # Developed at https://github.com/stackabletech/hadoop/tree/spike/override-datanode-id
+    git apply < /stackable/patches/0002-datanode-registration-override.patch && \
```

Comment:
Need a nicer way to maintain these patchsets... any ideas? @stackabletech/developers

Reply:
I asked the same question a few weeks ago and ended up with
https://github.com/stackabletech/docker-images/tree/main/hbase/stackable/patches
https://github.com/stackabletech/docker-images/blob/main/hbase/Dockerfile#L41-L42

Reply:
I think that's fine for applying them, but editing .patches directly isn't exactly a super friendly workflow either :/

Reply:
I've been looking for a solution for over 11 years
https://stackoverflow.com/questions/10705446/maintaining-multiple-patches-feature-branches-for-multiple-development-branches

Reply:
The patching mechanism for the Hadoop image was implemented in #526. I took the liberty of merging the main branch into this one.

Reply:
I'll get back to merging that once the rest of the PR is finalized.

Reply:
Merged now.